controller/user.js

```diff
@@ -42,10 +42,26 @@ exports.loginWithOtp = async (req, res) => {
         // Send OTP to user (e.g., via email or SMS)
         sendMail(req.body.email, otp);
         // save the secret key in the user object or database
-        user.secret = secret.base32;
+        await client.set(user.email, secret.base32, { EX: 300 }, (err, res) => {
+            if (err) console.log(err);
```

Comment:
If not saved, OTP shouldn't go. Handle error appropriately. Send email only after saving the secret